pkg/oc/cli/importer/appjson/appjson.go

```diff
@@ -201,40 +233,50 @@ func (o *AppJSONOptions) Run() error {
 		return err
 	}
 
-	template.ObjectLabels = map[string]string{"app.json": template.Name}
-
-	// all the types generated into the template should be known
-	if errs := app.AsVersionedObjects(template.Objects, legacyscheme.Scheme, legacyscheme.Scheme, o.OutputVersions...); len(errs) > 0 {
-		for _, err := range errs {
-			fmt.Fprintf(o.bulkAction.ErrOut, "error: %v\n", err)
-		}
+	externalTemplate := &templatev1.Template{}
```

Comment:
Any particular reason we can't switch the generator to external?

Reply:
The generator depends on a lot of [oc/lib/newapp/app helpers](https://github.com/openshift/origin/blob/master/pkg/oc/lib/newapp/appjson/appjson.go#L173).

We should switch those to use external versions (and all their callers) first